### PR TITLE
[MI-2288][server+webapp]: Differentiated payload for service-type and Created chip UI for displaying filters on subscription cards.

### PR DESCRIPTION
**Pull request by avas27JTG**
- Differentiated payload for service-type
- Added filed for repo name
- Created chip UI for displaying filters on subscription cards

Screenshots:
image

image

### CODE REVIEW

File: webapp/src/components/card/subscription/index.tsx [hunk 1]
```diff
@@ -61,10 +62,29 @@ const SubscriptionCard = ({handleDeleteSubscrption, subscriptionDetails: {channe
...
+                                    {pluginConstants.SVGIcons.cardFilter}
+                                </SVGWrapper>
+                            </div>
+                            <div className='card-chip-wrapper'>
+                                {
+                                    repositoryName && <Chip text={`Repository is ${repositoryName}`}/>
+                                }
+                                {
+                                    targetBranch && <Chip text={`Target Branch is ${targetBranch}`}/>
+                                }
```
Comment (tanisha-agarwal) [inline webapp/src/components/card/subscription/index.tsx:83]:
```suggestion
                                {
                                    <Chip text={ repositoryName  
                                              ? `Repository is ${repositoryName}` 
                                              : `Target Branch is ${targetBranch}`
                                          }
                                    />
                                }
```

**Review by tanisha-agarwal — APPROVED** [review]

File: webapp/src/components/card/subscription/index.tsx (hunk 1 above)
Comment (avas27JTG) [reply webapp/src/components/card/subscription/index.tsx:83]: If both `repositoryName` and `targetBranch` are present at a time then we need to show both of them, so we need two separate `Chip`. If we use ternary then only one of the two will be shown at a time.

Comment (avas27JTG) [reply webapp/src/components/card/subscription/index.tsx:83]: you can see attached screenshot for this UI case

**Review by raghavaggarwal2308 — APPROVED** [review]

File: webapp/src/components/chip/index.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import './styles.scss';
+
+type ChipProps = {
+    text: string
+    extraClass?: string
+}
+
+const Chip = ({text, extraClass = ''}: ChipProps) => (
+    <Tooltip tooltipContent={text}>
+        <div className={`chip Badge__box text-truncate ${extraClass}`}>{text}</div>
```
Comment (manojmalik20) [inline webapp/src/components/chip/index.tsx:14]: Why `Badge__box` class had different casing than other classes?

**Review by manojmalik20 — APPROVED** [review]

File: webapp/src/components/chip/index.tsx (hunk 2 above)
Comment (avas27JTG) [reply webapp/src/components/chip/index.tsx:14]: Yes this is inconsistent naming but it's from Mattermost webapp we have not created this, using directly from MM styling

**Pull request merged**